core/src/impl/Kokkos_ViewLayoutTiled.hpp

```diff
@@ -131,14 +131,14 @@ struct ViewOffset<
 
   enum { VORank = Dimension::rank };
 
-  enum { SHIFT_0 = Kokkos::Impl::integral_power_of_two(Layout::N0) };
-  enum { SHIFT_1 = Kokkos::Impl::integral_power_of_two(Layout::N1) };
-  enum { SHIFT_2 = Kokkos::Impl::integral_power_of_two(Layout::N2) };
-  enum { SHIFT_3 = Kokkos::Impl::integral_power_of_two(Layout::N3) };
-  enum { SHIFT_4 = Kokkos::Impl::integral_power_of_two(Layout::N4) };
-  enum { SHIFT_5 = Kokkos::Impl::integral_power_of_two(Layout::N5) };
-  enum { SHIFT_6 = Kokkos::Impl::integral_power_of_two(Layout::N6) };
-  enum { SHIFT_7 = Kokkos::Impl::integral_power_of_two(Layout::N7) };
+  enum : unsigned { SHIFT_0 = Kokkos::Impl::integral_power_of_two(Layout::N0) };
+  enum : unsigned { SHIFT_1 = Kokkos::Impl::integral_power_of_two(Layout::N1) };
+  enum : unsigned { SHIFT_2 = Kokkos::Impl::integral_power_of_two(Layout::N2) };
+  enum : unsigned { SHIFT_3 = Kokkos::Impl::integral_power_of_two(Layout::N3) };
+  enum : unsigned { SHIFT_4 = Kokkos::Impl::integral_power_of_two(Layout::N4) };
+  enum : unsigned { SHIFT_5 = Kokkos::Impl::integral_power_of_two(Layout::N5) };
+  enum : unsigned { SHIFT_6 = Kokkos::Impl::integral_power_of_two(Layout::N6) };
+  enum : unsigned { SHIFT_7 = Kokkos::Impl::integral_power_of_two(Layout::N7) };
```

Comment:
this would require C++17 since we inline initialize the stuff or (or have a duplicated declaration outside of the class). And constexpr definitely didn't work on XL with OpenMPTarget, and I am not sure what the OpenMP standard in general says about availability of symbols in the offloaded region for static constexpr. Static variables are generally NOT supported.